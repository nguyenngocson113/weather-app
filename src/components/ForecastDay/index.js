import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import SmallLabel from 'components/Label/SmallLabel';
import Text from 'components/Text';
import device from 'responsive/Device';

const ForecastWrapper = styled.div`
  flex-shrink: 0;
  flex-basis: 90px;
  padding: 10px;
  margin: 0 5px;
  border-radius: 5px;
  background-color: rgba(255, 255, 255, 0.2);
  &:first-child {
    margin-left: 0;
  }
  &:last-child {
    margin-right: 0;
  }
  @media ${device.tablet} {
    flex-basis: 110px;
  }
  @media ${device.laptop} {
    flex-basis: 125px;
  }
  @media ${device.laptopL} {
    flex-basis: 140px;
  }
`;

const WeatherIcon = styled.img`
  display: block;
  height: 50px;
  width: 50px;
  margin: 0 auto;
`;

const ForecastDay = props => {
  const { temp, date, icon } = props;
  const iconUrl = `https://openweathermap.org/img/w/${icon}.png`;

  return (
    <ForecastWrapper>
      <Text align="center">{date}</Text>
      <WeatherIcon src={iconUrl} />
      <SmallLabel align="center" weight="400">
        {temp}
      </SmallLabel>
    </ForecastWrapper>
  );
};

ForecastDay.propTypes = {
  temp: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default ForecastDay;
