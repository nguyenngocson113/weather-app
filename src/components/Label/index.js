export { default as BigLabel } from './BigLabel';
export { default as MediumLabel } from './MediumLabel';
export { default as SmallLabel } from './SmallLabel';
