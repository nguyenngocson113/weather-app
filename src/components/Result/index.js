import React from 'react';

// components
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faCloud,
  faBolt,
  faCloudRain,
  faCloudShowersHeavy,
  faSnowflake,
  faSun,
  faSmog,
} from '@fortawesome/free-solid-svg-icons';
import styled from 'styled-components';
import device from 'responsive/Device';
import ForecastDay from 'components/ForecastDay';
import ResultFadeIn from 'components/ResultFadeIn';
import { BigLabel, MediumLabel, SmallLabel } from 'components/Label';
import Text from 'components/Text';

// states
import { useWeatherState, useForecastState } from 'layout/weather/data-actions';

// constants
import { DAYS } from 'layout/weather/consts';

export const Results = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 40px 0;
  opacity: 0;
  visibility: hidden;
  position: relative;
  top: 20px;
  animation: ${ResultFadeIn} 0.5s 1.4s forwards;
`;

export const LocationWrapper = styled.div`
  flex-basis: 100%;
`;

export const CurrentWeatherWrapper = styled.div`
  flex-basis: 100%;
  display: grid;
  justify-content: center;
  align-items: center;
  grid-template-columns: auto 1fr;
  margin: 20px 0;
  grid-gap: 30px;
  @media ${device.mobileL} {
    flex-basis: 50%;
    padding-right: 10px;
  }
  @media ${device.tablet} {
    grid-template-columns: 1fr 1fr;
    padding-right: 20px;
  }
`;

export const WeatherIcon = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 70px;
  color: #ffffff;
  @media ${device.tablet} {
    font-size: 100px;
    justify-content: flex-end;
  }
  @media ${device.laptop} {
    font-size: 120px;
  }
  @media ${device.laptopL} {
    font-size: 140px;
  }
`;

export const TemperatureWrapper = styled.div``;

export const Temperature = styled.h3`
  display: block;
  font-size: 50px;
  font-weight: 400;
  color: #ffffff;
  @media ${device.tablet} {
    font-size: 70px;
  }
  @media ${device.laptop} {
    font-size: 90px;
  }
  @media ${device.laptopL} {
    font-size: 110px;
  }
`;

export const WeatherDetailsWrapper = styled.div`
  flex-basis: 100%;
  display: flex;
  flex-wrap: wrap;
  padding: 10px 0;
  margin: 20px 0;
  background-color: rgba(255, 255, 255, 0.2);
  border-radius: 10px;
  align-self: flex-start;
  @media ${device.mobileL} {
    flex-basis: 50%;
  }
`;

export const WeatherDetail = styled.div`
  flex-basis: calc(100% / 3);
  padding: 10px;
  @media ${device.laptop} {
    padding: 20px 10px;
  }
`;

export const ForecastWrapper = styled.div`
  flex-basis: 100%;
  margin: 20px 0;
  overflow: hidden;
`;

export const Forecast = styled.div`
  position: relative;
  display: flex;
  overflow-x: scroll;
  overflow-y: hidden;
  scrollbar-color: lightgray #ffffff;
  scrollbar-width: thin;
  margin-top: 20px;
  padding-bottom: 20px;
  margin: 0 auto;
  justify-content: start;
  @media ${device.laptop} {
    order: 4;
  }
  @media ${device.tablet} {
    justify-content: center;
  }
`;

const WEATHER_ICONS = {
  Thunderstorm: faBolt,
  Drizzle: faCloudRain,
  Rain: faCloudShowersHeavy,
  Snow: faSnowflake,
  Clear: faSun,
  Clouds: faCloud,
};

const Result = () => {
  const { weatherData } = useWeatherState();
  const { forecastData } = useForecastState();

  if (!weatherData || !forecastData) {
    return null;
  }

  const {
    city,
    country,
    date,
    description,
    main,
    temp,
    sunset,
    sunrise,
    humidity,
    wind,
    highestTemp,
    lowestTemp,
  } = weatherData;

  const getTemp = temp => `${Math.floor(temp * 1) / 1}°`;

  const forecasts = forecastData.map(item => {
    const date = new Date(item.dt * 1000);
    return (
      <ForecastDay
        key={item.dt}
        temp={`${getTemp(item.temp.max)} - ${getTemp(item.temp.min)}`}
        icon={item.weather[0].icon}
        date={DAYS[date.getDay()]}
      />
    );
  });

  const renderWeatherDetails = () => {
    const weatherParams = [
      { value: `${Math.floor(highestTemp)}°`, text: 'Hight' },
      { value: `${wind}mph`, text: 'Wind' },
      { value: sunrise, text: 'Sunrise' },
      { value: `${Math.floor(lowestTemp)}°`, text: 'Low' },
      { value: `${humidity}%`, text: 'Rain' },
      { value: sunset, text: 'Sunset' },
    ];
    return weatherParams.map(param => (
      <WeatherDetail key={param.text}>
        <SmallLabel align="center" weight="400">
          {param.value}
        </SmallLabel>
        <Text align="center">{param.text}</Text>
      </WeatherDetail>
    ));
  };

  return (
    <Results>
      <LocationWrapper>
        <BigLabel>
          {city}, {country}
        </BigLabel>
        <SmallLabel weight="400">{date}</SmallLabel>
      </LocationWrapper>
      <CurrentWeatherWrapper>
        <WeatherIcon>
          <FontAwesomeIcon icon={WEATHER_ICONS[main] || faSmog} />
        </WeatherIcon>
        <TemperatureWrapper>
          <Temperature>{Math.floor(temp)}&#176;</Temperature>
          <SmallLabel weight="400" firstToUpperCase>
            {description}
          </SmallLabel>
        </TemperatureWrapper>
      </CurrentWeatherWrapper>
      <WeatherDetailsWrapper>{renderWeatherDetails()}</WeatherDetailsWrapper>
      <ForecastWrapper>
        <MediumLabel weight="400">Forecast</MediumLabel>
        <Forecast>{forecasts}</Forecast>
      </ForecastWrapper>
    </Results>
  );
};

export default Result;
