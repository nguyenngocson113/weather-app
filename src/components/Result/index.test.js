import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import * as Redux from 'react-redux';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import weatherMockData from './__test__/mock-data-weather.json';
import forecastMockData from './__test__/mock-data-forecast.json';
import Result, {
  LocationWrapper,
  CurrentWeatherWrapper,
  WeatherIcon,
  TemperatureWrapper,
  Temperature,
  WeatherDetailsWrapper,
  WeatherDetail,
  ForecastWrapper,
  Forecast,
} from './index';
import { BigLabel, SmallLabel, MediumLabel } from 'components/Label';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Text from 'components/Text';
import ForecastDay from 'components/ForecastDay';

configure({ adapter: new Adapter() });
describe('<Result /> is rendered', () => {
  let useSelectorSpy;

  beforeEach(() => {
    useSelectorSpy = jest.spyOn(Redux, 'useSelector');
  });

  test('return null when dont have weather and forecast', () => {
    useSelectorSpy.mockReturnValue({});
    const wrapper = renderer.create(<Result />).toJSON();
    expect(wrapper).toEqual(null);
  });
});

describe('<Result /> is rendered', () => {
  let useSelectorSpy;

  beforeEach(() => {
    useSelectorSpy = jest.spyOn(Redux, 'useSelector');
    useSelectorSpy.mockReturnValue({
      weatherData: weatherMockData,
      forecastData: forecastMockData,
    });
  });

  test('snapshot <Result /> component', () => {
    const wrapper = renderer.create(<Result />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  test('with style of Results tag', () => {
    const wrapper = shallow(<Result />);
    expect(wrapper).toHaveStyleRule('display', 'flex');
    expect(wrapper).toHaveStyleRule('flex-wrap', 'wrap');
    expect(wrapper).toHaveStyleRule('justify-content', 'space-between');
    expect(wrapper).toHaveStyleRule('padding', '40px 0');
    expect(wrapper).toHaveStyleRule('opacity', '0');
    expect(wrapper).toHaveStyleRule('visibility', 'hidden');
    expect(wrapper).toHaveStyleRule('top', '20px');
  });

  describe('<LocationWrapper /> is rendered', () => {
    test('with style of LocationWrapper tag', () => {
      const wrapper = shallow(<Result />);
      const location = wrapper.find(LocationWrapper);
      expect(location.exists()).toEqual(true);
      expect(location).toHaveStyleRule('flex-basis', '100%');
    });
    test('with <BigLabel />', () => {
      const wrapper = shallow(<Result />);
      const location = wrapper.find(LocationWrapper);
      const bigLabel = location.find(BigLabel);
      expect(bigLabel.exists()).toEqual(true);
      expect(bigLabel.text()).toEqual(`${weatherMockData.city}, ${weatherMockData.country}`);
    });
    test('with <SmallLabel />', () => {
      const wrapper = shallow(<Result />);
      const location = wrapper.find(LocationWrapper);
      const smallLabel = location.find(SmallLabel);
      expect(smallLabel.exists()).toEqual(true);
      expect(smallLabel.text()).toEqual(`${weatherMockData.date}`);
    });
  });

  describe('<CurrentWeatherWrapper /> is rendered', () => {
    test('with style of CurrentWeatherWrapper tag', () => {
      const wrapper = shallow(<Result />);
      const currentWeather = wrapper.find(CurrentWeatherWrapper);
      expect(currentWeather.exists()).toEqual(true);
      expect(currentWeather).toHaveStyleRule('flex-basis', '100%');
      expect(currentWeather).toHaveStyleRule('display', 'grid');
      expect(currentWeather).toHaveStyleRule('justify-content', 'center');
      expect(currentWeather).toHaveStyleRule('align-items', 'center');
      expect(currentWeather).toHaveStyleRule('grid-template-columns', 'auto 1fr');
      expect(currentWeather).toHaveStyleRule('margin', '20px 0');
      expect(currentWeather).toHaveStyleRule('grid-gap', '30px');
    });
    test('with <WeatherIcon />', () => {
      const wrapper = shallow(<Result />);
      const currentWeather = wrapper.find(CurrentWeatherWrapper);
      const weatherIcon = currentWeather.find(WeatherIcon);
      expect(weatherIcon.exists()).toEqual(true);
      expect(weatherIcon.find(FontAwesomeIcon).exists()).toEqual(true);
    });
    test('with <TemperatureWrapper />', () => {
      const wrapper = shallow(<Result />);
      const currentWeather = wrapper.find(CurrentWeatherWrapper);
      const temperature = currentWeather.find(TemperatureWrapper);
      expect(temperature.exists()).toEqual(true);
    });
    test('with <Temperature />', () => {
      const wrapper = shallow(<Result />);
      const currentWeather = wrapper.find(CurrentWeatherWrapper);
      const temperatureWrap = currentWeather.find(TemperatureWrapper);
      const temp = temperatureWrap.find(Temperature);
      expect(temp.exists()).toEqual(true);
      expect(temp.text()).toEqual(`${Math.floor(weatherMockData.temp)}°`);
      expect(temp).toHaveStyleRule('display', 'block');
      expect(temp).toHaveStyleRule('font-size', '50px');
      expect(temp).toHaveStyleRule('font-weight', '400');
      expect(temp).toHaveStyleRule('color', '#ffffff');
    });
    test('with <SmallLabel />', () => {
      const wrapper = shallow(<Result />);
      const currentWeather = wrapper.find(CurrentWeatherWrapper);
      const temperatureWrap = currentWeather.find(TemperatureWrapper);
      const smallLabel = temperatureWrap.find(SmallLabel);
      expect(smallLabel.exists()).toEqual(true);
      expect(smallLabel.text()).toEqual(`${weatherMockData.description}`);
    });
  });

  describe('<WeatherDetailsWrapper /> is rendered', () => {
    let renderWeatherDetails;
    beforeEach(() => {
      renderWeatherDetails = jest.fn(x => {});
    });
    test('with style of WeatherDetailsWrapper tag', () => {
      const wrapper = shallow(<Result />);
      const currentWeather = wrapper.find(WeatherDetailsWrapper);
      expect(currentWeather.exists()).toEqual(true);
      expect(currentWeather).toHaveStyleRule('flex-basis', '100%');
      expect(currentWeather).toHaveStyleRule('flex-wrap', 'wrap');
      expect(currentWeather).toHaveStyleRule('padding', '10px 0');
      expect(currentWeather).toHaveStyleRule('margin', '20px 0');
      expect(currentWeather).toHaveStyleRule('background-color', 'rgba(255,255,255,0.2)');
      expect(currentWeather).toHaveStyleRule('border-radius', '10px');
      expect(currentWeather).toHaveStyleRule('align-self', 'flex-start');
    });

    describe('List <WeatherDetailsWrapper /> is rendered', () => {
      let wrapper;
      let currentWeather;
      let details;

      beforeEach(() => {
        wrapper = shallow(<Result />);
        currentWeather = wrapper.find(WeatherDetailsWrapper);
        details = currentWeather.find(WeatherDetail);
      });
      test('having 6 number items', () => {
        expect(details.exists()).toEqual(true);
        expect(details).toHaveLength(6);
        // expect(details).toMatchSnapshot();
      });

      test('First <WeatherDetail /> is rendered with Hight', () => {
        const hightComp = details.at(0);
        expect(hightComp.find(SmallLabel).exists()).toEqual(true);
        expect(hightComp.find(SmallLabel).text()).toEqual(
          `${Math.floor(weatherMockData.highestTemp)}°`,
        );
        expect(hightComp.find(Text).text()).toEqual('Hight');
      });
      test('First <WeatherDetail /> is rendered with Wind', () => {
        const windComp = details.at(1);
        expect(windComp.find(SmallLabel).exists()).toEqual(true);
        expect(windComp.find(SmallLabel).text()).toEqual(`${weatherMockData.wind}mph`);
        expect(windComp.find(Text).text()).toEqual('Wind');
      });
      test('First <WeatherDetail /> is rendered with Sunrise', () => {
        const sunriseComp = details.at(2);
        expect(sunriseComp.find(SmallLabel).exists()).toEqual(true);
        expect(sunriseComp.find(SmallLabel).text()).toEqual(weatherMockData.sunrise);
        expect(sunriseComp.find(Text).text()).toEqual('Sunrise');
      });
      test('First <WeatherDetail /> is rendered with Low', () => {
        const lowComp = details.at(3);
        expect(lowComp.find(SmallLabel).exists()).toEqual(true);
        expect(lowComp.find(SmallLabel).text()).toEqual(
          `${Math.floor(weatherMockData.lowestTemp)}°`,
        );
        expect(lowComp.find(Text).text()).toEqual('Low');
      });
      test('First <WeatherDetail /> is rendered with Rain', () => {
        const rainComp = details.at(4);
        expect(rainComp.find(SmallLabel).exists()).toEqual(true);
        expect(rainComp.find(SmallLabel).text()).toEqual(`${weatherMockData.humidity}%`);
        expect(rainComp.find(Text).text()).toEqual('Rain');
      });
      test('First <WeatherDetail /> is rendered with Sunset', () => {
        const sunsetComp = details.at(5);
        expect(sunsetComp.find(SmallLabel).exists()).toEqual(true);
        expect(sunsetComp.find(SmallLabel).text()).toEqual(weatherMockData.sunset);
        expect(sunsetComp.find(Text).text()).toEqual('Sunset');
      });
    });

    describe('<ForecastWrapper /> is rendered', () => {
      test('with style of CurrentWeatherWrapper tag', () => {
        const wrapper = shallow(<Result />);
        const forecastWrapper = wrapper.find(ForecastWrapper);
        expect(forecastWrapper.exists()).toEqual(true);
        expect(forecastWrapper).toHaveStyleRule('flex-basis', '100%');
        expect(forecastWrapper).toHaveStyleRule('margin', '20px 0');
        expect(forecastWrapper).toHaveStyleRule('overflow', 'hidden');
      });
      test('with <MediumLabel />', () => {
        const wrapper = shallow(<Result />);
        const forecastWrapper = wrapper.find(ForecastWrapper);
        const mediumLabel = forecastWrapper.find(MediumLabel);
        expect(mediumLabel.text()).toEqual('Forecast');
      });
      test('with <Forecast /> styles', () => {
        const wrapper = shallow(<Result />);
        const forecastWrapper = wrapper.find(ForecastWrapper);
        const forecast = forecastWrapper.find(Forecast);
        expect(forecast.exists()).toEqual(true);
        expect(forecast).toHaveStyleRule('position', 'relative');
        expect(forecast).toHaveStyleRule('display', 'flex');
        expect(forecast).toHaveStyleRule('overflow-x', 'scroll');
        expect(forecast).toHaveStyleRule('overflow-y', 'hidden');
        expect(forecast).toHaveStyleRule('scrollbar-color', 'lightgray #ffffff');
        expect(forecast).toHaveStyleRule('scrollbar-width', 'thin');
        expect(forecast).toHaveStyleRule('margin-top', '20px');
        expect(forecast).toHaveStyleRule('padding-bottom', '20px');
        expect(forecast).toHaveStyleRule('margin', '0 auto');
        expect(forecast).toHaveStyleRule('justify-content', 'start');
      });

      test('having 5 number items from current day', () => {
        const wrapper = shallow(<Result />);
        const forecastWrapper = wrapper.find(ForecastWrapper);
        const forecast = forecastWrapper.find(Forecast).find(ForecastDay);
        expect(forecast.exists()).toEqual(true);
        expect(forecast).toHaveLength(5);
      });
    });
  });
});
