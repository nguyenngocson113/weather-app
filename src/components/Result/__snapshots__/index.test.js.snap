// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> is rendered snapshot <Result /> component 1`] = `
.c3 {
  color: #FFFFFF;
  display: block;
  font-weight: 400;
  font-size: 15px;
  text-align: left;
  padding: 5px 0;
}

.c7 {
  color: #FFFFFF;
  display: block;
  font-weight: 400;
  font-size: 15px;
  text-align: left;
  padding: 5px 0;
}

.c7:first-letter {
  text-transform: uppercase;
}

.c10 {
  color: #FFFFFF;
  display: block;
  font-weight: 400;
  font-size: 15px;
  text-align: center;
  padding: 5px 0;
}

.c11 {
  color: #FFFFFF;
  display: block;
  font-size: 12px;
  text-align: center;
}

.c15 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 90px;
  -ms-flex-preferred-size: 90px;
  flex-basis: 90px;
  padding: 10px;
  margin: 0 5px;
  border-radius: 5px;
  background-color: rgba(255,255,255,0.2);
}

.c15:first-child {
  margin-left: 0;
}

.c15:last-child {
  margin-right: 0;
}

.c16 {
  display: block;
  height: 50px;
  width: 50px;
  margin: 0 auto;
}

.c2 {
  color: #FFFFFF;
  display: block;
  font-weight: 600;
  font-size: 30px;
  text-align: left;
  padding: 5px 0;
}

.c13 {
  color: #FFFFFF;
  display: block;
  font-weight: 400;
  font-size: 20px;
  text-align: left;
  padding: 5px 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 40px 0;
  opacity: 0;
  visibility: hidden;
  position: relative;
  top: 20px;
  -webkit-animation: dhgPis 0.5s 1.4s forwards;
  animation: dhgPis 0.5s 1.4s forwards;
}

.c1 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
}

.c4 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-columns: auto 1fr;
  margin: 20px 0;
  grid-gap: 30px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 70px;
  color: #ffffff;
}

.c6 {
  display: block;
  font-size: 50px;
  font-weight: 400;
  color: #ffffff;
}

.c8 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 10px 0;
  margin: 20px 0;
  background-color: rgba(255,255,255,0.2);
  border-radius: 10px;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c9 {
  -webkit-flex-basis: calc(100% / 3);
  -ms-flex-preferred-size: calc(100% / 3);
  flex-basis: calc(100% / 3);
  padding: 10px;
}

.c12 {
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  margin: 20px 0;
  overflow: hidden;
}

.c14 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow-x: scroll;
  overflow-y: hidden;
  -webkit-scrollbar-color: lightgray #ffffff;
  -moz-scrollbar-color: lightgray #ffffff;
  -ms-scrollbar-color: lightgray #ffffff;
  scrollbar-color: lightgray #ffffff;
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  margin-top: 20px;
  padding-bottom: 20px;
  margin: 0 auto;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

@media (min-width:768px) {
  .c3 {
    font-size: 20px;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 23px;
  }
}

@media (min-width:1440px) {
  .c3 {
    font-size: 26px;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 20px;
  }
}

@media (min-width:1024px) {
  .c7 {
    font-size: 23px;
  }
}

@media (min-width:1440px) {
  .c7 {
    font-size: 26px;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 20px;
  }
}

@media (min-width:1024px) {
  .c10 {
    font-size: 23px;
  }
}

@media (min-width:1440px) {
  .c10 {
    font-size: 26px;
  }
}

@media (min-width:768px) {
  .c11 {
    font-size: 15px;
  }
}

@media (min-width:1024px) {
  .c11 {
    font-size: 17px;
  }
}

@media (min-width:1440px) {
  .c11 {
    font-size: 19px;
  }
}

@media (min-width:768px) {
  .c15 {
    -webkit-flex-basis: 110px;
    -ms-flex-preferred-size: 110px;
    flex-basis: 110px;
  }
}

@media (min-width:1024px) {
  .c15 {
    -webkit-flex-basis: 125px;
    -ms-flex-preferred-size: 125px;
    flex-basis: 125px;
  }
}

@media (min-width:1440px) {
  .c15 {
    -webkit-flex-basis: 140px;
    -ms-flex-preferred-size: 140px;
    flex-basis: 140px;
  }
}

@media (min-width:768px) {
  .c2 {
    font-size: 37px;
  }
}

@media (min-width:1024px) {
  .c2 {
    font-size: 43px;
  }
}

@media (min-width:1440px) {
  .c2 {
    font-size: 52px;
  }
}

@media (min-width:768px) {
  .c13 {
    font-size: 23px;
  }
}

@media (min-width:1024px) {
  .c13 {
    font-size: 26px;
  }
}

@media (min-width:1440px) {
  .c13 {
    font-size: 29px;
  }
}

@media (min-width:525px) {
  .c4 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    padding-right: 10px;
  }
}

@media (min-width:768px) {
  .c4 {
    grid-template-columns: 1fr 1fr;
    padding-right: 20px;
  }
}

@media (min-width:768px) {
  .c5 {
    font-size: 100px;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

@media (min-width:1024px) {
  .c5 {
    font-size: 120px;
  }
}

@media (min-width:1440px) {
  .c5 {
    font-size: 140px;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 70px;
  }
}

@media (min-width:1024px) {
  .c6 {
    font-size: 90px;
  }
}

@media (min-width:1440px) {
  .c6 {
    font-size: 110px;
  }
}

@media (min-width:525px) {
  .c8 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media (min-width:1024px) {
  .c9 {
    padding: 20px 10px;
  }
}

@media (min-width:1024px) {
  .c14 {
    -webkit-order: 4;
    -ms-flex-order: 4;
    order: 4;
  }
}

@media (min-width:768px) {
  .c14 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <h2
      className="c2"
    >
      Vinh
      , 
      VN
    </h2>
    <h4
      className="c3"
    >
      Monday 7 December
    </h4>
  </div>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cloud fa-w-20 "
        data-icon="cloud"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M537.6 226.6c4.1-10.7 6.4-22.4 6.4-34.6 0-53-43-96-96-96-19.7 0-38.1 6-53.3 16.2C367 64.2 315.3 32 256 32c-88.4 0-160 71.6-160 160 0 2.7.1 5.4.2 8.1C40.2 219.8 0 273.2 0 336c0 79.5 64.5 144 144 144h368c70.7 0 128-57.3 128-128 0-61.9-44-113.6-102.4-125.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className=""
    >
      <h3
        className="c6"
      >
        19
        °
      </h3>
      <h4
        className="c7"
      >
        overcast clouds
      </h4>
    </div>
  </div>
  <div
    className="c8"
  >
    <div
      className="c9"
    >
      <h4
        className="c10"
      >
        19°
      </h4>
      <span
        className="c11"
      >
        Hight
      </span>
    </div>
    <div
      className="c9"
    >
      <h4
        className="c10"
      >
        4.53mph
      </h4>
      <span
        className="c11"
      >
        Wind
      </span>
    </div>
    <div
      className="c9"
    >
      <h4
        className="c10"
      >
        6:17:
      </h4>
      <span
        className="c11"
      >
        Sunrise
      </span>
    </div>
    <div
      className="c9"
    >
      <h4
        className="c10"
      >
        19°
      </h4>
      <span
        className="c11"
      >
        Low
      </span>
    </div>
    <div
      className="c9"
    >
      <h4
        className="c10"
      >
        78%
      </h4>
      <span
        className="c11"
      >
        Rain
      </span>
    </div>
    <div
      className="c9"
    >
      <h4
        className="c10"
      >
        5:20:
      </h4>
      <span
        className="c11"
      >
        Sunset
      </span>
    </div>
  </div>
  <div
    className="c12"
  >
    <h3
      className="c13"
    >
      Forecast
    </h3>
    <div
      className="c14"
    >
      <div
        className="c15"
      >
        <span
          className="c11"
        >
          Monday
        </span>
        <img
          className="c16"
          src="https://openweathermap.org/img/w/04d.png"
        />
        <h4
          className="c10"
        >
          20° - 17°
        </h4>
      </div>
      <div
        className="c15"
      >
        <span
          className="c11"
        >
          Tuesday
        </span>
        <img
          className="c16"
          src="https://openweathermap.org/img/w/04d.png"
        />
        <h4
          className="c10"
        >
          20° - 16°
        </h4>
      </div>
      <div
        className="c15"
      >
        <span
          className="c11"
        >
          Wednesday
        </span>
        <img
          className="c16"
          src="https://openweathermap.org/img/w/03d.png"
        />
        <h4
          className="c10"
        >
          22° - 15°
        </h4>
      </div>
      <div
        className="c15"
      >
        <span
          className="c11"
        >
          Thursday
        </span>
        <img
          className="c16"
          src="https://openweathermap.org/img/w/10d.png"
        />
        <h4
          className="c10"
        >
          21° - 17°
        </h4>
      </div>
      <div
        className="c15"
      >
        <span
          className="c11"
        >
          Friday
        </span>
        <img
          className="c16"
          src="https://openweathermap.org/img/w/10d.png"
        />
        <h4
          className="c10"
        >
          24° - 19°
        </h4>
      </div>
    </div>
  </div>
</div>
`;
